File: CS/ViewModel.cs
using DevExpress.Mvvm;
using System.Collections.ObjectModel;

namespace AutoPopulateColumns {
    public class ViewModel : ViewModelBase {
        public ViewModel() {
            Source = DataModel.GetCustomers();
        }
        public ObservableCollection<Customer> Source { get; }
    }
}

File: VB/ViewModel.vb
Imports DevExpress.Mvvm
Imports System.Collections.ObjectModel

Namespace AutoPopulateColumns

    Public Class ViewModel
        Inherits ViewModelBase

        Public Sub New()
            Source = DataModel.GetCustomers()
        End Sub

        Public ReadOnly Property Source As ObservableCollection(Of Customer)
    End Class
End Namespace
